### PR TITLE
Add support for  "where in" queries

### PR DESCRIPTION
**Pull request by jreadey**
WIP review...

### CODE REVIEW

File: hsds/util/chunkUtil.py [hunk 1]
```diff
@@ -113,9 +114,8 @@ def expandChunk(
                 else:
                     pass  # can't extend chunk along this dimension
         if chunk_size <= old_chunk_size:
-            # reality check to see if we'll ever break out of the while loop
-            log.warn("Unexpected error in guess_chunk size")
-
+            # stop iteration if we haven't increased the chunk size
+            log.debug("stopping epandChunk iteration")
```
Comment (mattjala) [inline hsds/util/chunkUtil.py:118]: `epandChunk` -> `expandChunk`

File: hsds/util/chunkUtil.py [hunk 2]
```diff
@@ -1148,41 +1273,95 @@ def chunkQuery(
...
+    if where_field:
+        log.debug(f"where_field: {where_field}")
+        if where_field not in field_names:
+            msg = f"where field {where_field} is not a member of dataset type"
+            raise ValueError(msg)
+        where_elements = _getWhereElements(query)
+        if not where_elements:
+            msg = "query: where key word with no elements"
+            raise ValueError(msg)
+        # convert to ndarray, checking than we can convert to our dtype along the way
```
Comment (mattjala) [inline hsds/util/chunkUtil.py:1308]: `than we can convert` -> `that we can convert`

File: hsds/util/chunkUtil.py [hunk 3]
```diff
@@ -1148,41 +1273,95 @@ def chunkQuery(
...
     else:
         replace_mask = None
 
-    x = chunk_arr[slices]
-    eval_str = _getEvalStr(query, "x", field_names)
+    if not eval_str:
+        if where_field:
+            # return array based on where in elements
+            log.debug("query - no eval_str, returning where in array")
+            return chunk_arr
```
Comment (mattjala) [inline hsds/util/chunkUtil.py:1360]: `chunk_sel` seems to be the result of the selection from a `where in`  query - should that be returned here instead?

File: docs/requirements-rtd.txt [hunk 4]
```diff
@@ -0,0 +1,4 @@
+sphinx==5.0.0
+sphinx_rtd_theme==1.0.0
+readthedocs-sphinx-search==0.1.1
```
Comment (mattjala) [inline docs/requirements-rtd.txt:3]: Version for `readthedocs-sphinx-search` should be bumped to at least 0.3.2 to resolve the security issue

File: docs/advanced/authorization.rst [hunk 5]
```diff
@@ -0,0 +1,230 @@
+Authorization and Authentication
+================================
+
+Request Authentication
+----------------------
+
+The HSDS REST supports HTTP Basic authentication to authenticate users
```
Comment (mattjala) [inline docs/advanced/authorization.rst:7]: `HSDS REST` -> `HSDS REST API`

File: docs/advanced/authorization.rst [hunk 6]
```diff
@@ -0,0 +1,230 @@
...
+by comparing an encrypted username and password against a value stored
+within a password file. Alternatively, Bearer authentication can be used
+where a security token is issued by a provider such as Active Directory.
+In either case, once the request is authenticated a requestor username
+will be determine (unless anonymous request are supported, in which case
```
Comment (mattjala) [inline docs/advanced/authorization.rst:12]: `determine` -> `determined`

File: docs/advanced/authorization.rst [hunk 7]
```diff
@@ -0,0 +1,230 @@
...
+-  PUT /datasets/<id>/attributes/<name> - denied (returns HTTP Status
+   403 - Forbidden)
+-  DELETE /datasets/<id> - denied (returns HTTP Status 403 - Forbidden)
+
+Finally the same set of requests are sent with ‘ann’ as the user:
+
+-  GET /datasets/<id> - granted (returns HTTP Status 200 - OK)
+-  POST /datasets/<id>/value - granted (returns HTTP Status 200 - OK)
+-  PUT /datasets/<id>/shape) - grant (returns HTTP Status 200 - OK)
+-  PUT /datasets/<id>/attributes/<name> - denied (returns HTTP Status
```
Comment (mattjala) [inline docs/advanced/authorization.rst:115]: PUT and DELETE here should be 'granted'

File: docs/advanced/authorization.rst [hunk 8]
```diff
@@ -0,0 +1,230 @@
...
+
+The same response would be returned for an authenticated request where
+the user is neither ‘joe’ or ‘ann’.
+
+Next the same set of requests are sent with ‘joe’ as the user in the
+HTTP Authorization header:
+
+-  GET /datasets/<id> - granted (returns HTTP Status 200 - OK)
+-  POST /datasets/<id>/value - granted (returns HTTP Status 200 - OK)
+-  PUT /datasets/<id>/shape) - grant (returns HTTP Status 200 - OK)
```
Comment (mattjala) [inline docs/advanced/authorization.rst:105]: `grant` -> `granted`

File: docs/advanced/authorization.rst [hunk 9]
```diff
@@ -0,0 +1,230 @@
...
+-  PUT /datasets/<id>/attributes/<name> - denied (returns HTTP Status
+   403 - Forbidden)
+-  DELETE /datasets/<id> - denied (returns HTTP Status 403 - Forbidden)
+
+Finally the same set of requests are sent with ‘ann’ as the user:
+
+-  GET /datasets/<id> - granted (returns HTTP Status 200 - OK)
+-  POST /datasets/<id>/value - granted (returns HTTP Status 200 - OK)
+-  PUT /datasets/<id>/shape) - grant (returns HTTP Status 200 - OK)
```
Comment (mattjala) [inline docs/advanced/authorization.rst:114]: `grant` -> `granted`

File: docs/advanced/authorization.rst [hunk 10]
```diff
@@ -0,0 +1,230 @@
...
+   ========   ====  ======   ======  ======  =======  ========
+
+and that ‘ann’ and ‘joe’ are both members of the ‘devs’ group.
+
+The following requests on the domain sent with ‘joe’ as the user would
+be authorized or denied as follows:
+
+-  GET /datasets/<id> - granted (returns HTTP Status 200 - OK)
+-  POST /datasets/<id>/value - granted (returns HTTP Status 200 - OK)
+-  PUT /datasets/<id>/shape) - grant (returns HTTP Status 200 - OK)
```
Comment (mattjala) [inline docs/advanced/authorization.rst:199]: `grant` -> `granted`

File: docs/advanced/authorization.rst [hunk 11]
```diff
@@ -0,0 +1,230 @@
...
+
+If anonymous requests are supported (“allow_noauth” config value is
+True), the permissions would be the same as for authenticated users who
+are not “joe” or “ann”.
+
+For example, the following unauthenticated (no HTTP Authorization
+header) requests on the domain would be granted or denied as follows
+
+-  GET /datasets/<id> - granted (returns HTTP Status 200 - OK)
+-  POST /datasets/<id>/value - granted (returns HTTP Status 200 - OK)
```
[the remark this answers is not in the document]
Comment (mattjala) [reply docs/advanced/authorization.rst:89]: Actually, I think this should be `POST /datasets` if it's supposed to represent creation permissions

File: docs/advanced/frontdoor_install_azure.rst [hunk 12]
```diff
@@ -0,0 +1,94 @@
+.. _frontdoor_install:
+
+Frontdoor Installation
+=======================
+
+Use Azure Front Door for SSL/https with HSDS on AKS
+
+With the instructions in *kubernetes_install_azure.md*, you can deploy
+hsds to Azure Kubernetes Service (AKS) and access it over http using the
```
Comment (mattjala) [inline docs/advanced/frontdoor_install_azure.rst:9]: `hsds` -> `HSDS`

File: docs/contributing.rst [hunk 13]
```diff
@@ -0,0 +1,219 @@
+Bug Reports & Contributions
```
Comment (mattjala) [inline docs/contributing.rst:1]: This whole doc still refers to h5py

File: docs/faq.rst [hunk 14]
```diff
@@ -0,0 +1,220 @@
+.. _faq:
```
Comment (mattjala) [inline docs/faq.rst:1]: Another h5py doc

File: docs/licenses.rst [hunk 15]
```diff
@@ -0,0 +1,230 @@
+Licenses and legal info
```
Comment (mattjala) [inline docs/licenses.rst:1]: h5py document

File: docs/quick.rst [hunk 16]
```diff
@@ -0,0 +1,35 @@
+.. _quick:
```
Comment (mattjala) [inline docs/quick.rst:1]: h5py doc

File: docs/related_projects.rst [hunk 17]
```diff
@@ -0,0 +1,107 @@
+.. _related_projects:
```
Comment (mattjala) [inline docs/related_projects.rst:1]: h5py doc

File: docs/release_guide.rst [hunk 18]
```diff
@@ -0,0 +1,44 @@
+.. _release_guide:
```
Comment (mattjala) [inline docs/release_guide.rst:1]: h5py doc

File: docs/restapi/openapi.yaml [hunk 19]
```diff
@@ -0,0 +1,371 @@
...
+  - name: Link
+    description: Operations involving Links.
+  - name: Dataset
+    description: Operations involving Datasets.
+  - name: Datatype
+    description: Operations involving Datatypes.
+  - name: Attribute
+    description: Operations involving Attributes.
+  - name: ACLS
+    description: Operations involving Attributes.
```
Comment (mattjala) [inline docs/restapi/openapi.yaml:34]: `Attributes` -> `Access Control Lists`

File: docs/whatsnew/index.rst [hunk 20]
```diff
@@ -0,0 +1,11 @@
+.. _whatsnew:
```
Comment (mattjala) [inline docs/whatsnew/index.rst:1]: h5py doc

**Comment by jreadey** [comment]
I messed this up - didn't me to include commits from different branches.  Let me try to create a new PR...

**Pull request closed without merging**